packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -39,41 +43,64 @@ export class AtomicSearchInterface {
   @Prop() sample = false;
   @Prop({reflect: true}) pipeline = 'default';
   @Prop({reflect: true}) searchHub = 'default';
-  @Prop() logLevel?: LogLevel = 'silent';
+  @Prop() logLevel?: LogLevel;
   @Prop() i18n: i18n = i18next.createInstance();
+  @Prop({reflect: true}) language = 'en'; // TODO: make watchable and update i18next language on change
   @Prop({mutable: true}) engine?: Engine;
   @State() error?: Error;
+  @Event() ready!: EventEmitter;
 
   private unsubscribe: Unsubscribe = () => {};
   private hangingComponentsInitialization: InitializeEvent[] = [];
-  private initialized = false;
-  private afterInitializationCallbacks: (() => void)[] = [];
+  private systemsToInitialize = {
```

Comment:
if we need to init something else, we will only have to add it here, instead of making a new variable to check. When all of those are true, that's when the search interface is "ready"